hops/featurestore.py

```diff
@@ -0,0 +1,2651 @@
+                             stat_columns=None, num_bins=20, num_clusters=5,
+                             corr_method='pearson'):
+    """
+    Helper function that computes statistics of a featuregroup or training dataset using spark
+
+    Args:
+        name: the featuregroup or training dataset to update statistics for
+        spark_df: If a spark df is provided it will be used to compute statistics, otherwise the dataframe of the featuregroup will be fetched dynamically from the featurestore
+        version: the version of the featuregroup/training dataset (defaults to 1)
+        featurestore: the featurestore where the featuregroup or training dataset resides
+        descriptive_statistics: a boolean flag whether to compute descriptive statistics (min,max,mean etc) for the featuregroup/training dataset
+        feature_correlation: a boolean flag whether to compute a feature correlation matrix for the numeric columns in the featuregroup/training dataset
+        feature_histograms: a boolean flag whether to compute histograms for the numeric columns in the featuregroup/training dataset
+        cluster_analysis: a boolean flag whether to compute cluster analysis for the numeric columns in the featuregroup/training dataset
+        stat_columns: a list of columns to compute statistics for (defaults to all columns that are numeric)
+        num_bins: number of bins to use for computing histograms
+        num_clusters: the number of clusters to use for cluster analysis (k-means)
+        corr_method: the method to compute feature correlation with (pearson or spearman)
+
+    Returns:
+        feature_corr_data, desc_stats_data, features_histograms_data, cluster_analysis
+
+    """
+    if featurestore is None:
+        featurestore = project_featurestore()
+    if spark_df is None:
+        spark_df = get_featuregroup(name, featurestore, version)
+    if not stat_columns is None:
+        spark_df = spark_df.select(stat_columns)
+    feature_corr_data = None
+    desc_stats_data = None
+    features_histograms_data = None
+    cluster_analysis_data = None
+    spark = util._find_spark()
+
+    if spark_df.rdd.isEmpty():
+        raise AssertionError("Cannot compute statistics on an empty dataframe, the provided dataframe is empty")
+
+    if descriptive_statistics:
+        try:
+            print("computing descriptive statistics for : {}".format(name))
+            spark.sparkContext.setJobGroup("Descriptive Statistics Computation",
+                                           "Analyzing Dataframe Statistics for : {}".format(name))
+            desc_stats_json = _compute_descriptive_statistics(spark_df)
+            desc_stats_data = _structure_descriptive_stats_json(desc_stats_json)
+            spark.sparkContext.setJobGroup("", "")
+        except Exception as e:
+            raise AssertionError(
+                "Could not compute descriptive s
```

Comment:
@robzor92 I forgot to add, the notebook was tested with python 3.6 and python 2.7. This code snippet was the only versio-issue that I found

Reply:
(Line 1478)